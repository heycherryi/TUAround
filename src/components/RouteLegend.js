import React from 'react';
import PropTypes from 'prop-types';

const routeFrequency = [
  {
    name: 'Frequent',
    color: 'rgb(0, 233, 54)',
    textColor: 'rgb(265, 265, 265)',
  },
  {
    name: 'Local',
    color: 'rgb(117, 60, 190)',
    textColor: 'rgb(255, 255, 255)',
  },
  {
    name: 'Connector',
    color: 'rgb(93, 95, 98)',
    textColor: 'rgb(255, 255, 255)',
  },
  {
    name: 'Express',
    color: 'rgb(0, 170, 231)',
    textColor: 'rgb(255, 255, 255)',
  },
  {
    name: 'Train Shuttles',
    color: 'rgb(0, 30, 97)',
    textColor: 'rgb(255, 255, 255)',
  },
];

function RouteLegend() {
  return (
    <div className="d-flex mb-3 justify-content-center flex-wrap">
      {routeFrequency.map((frequency) => (
          <div key={frequency.name} className="mb-1 ms-1 me-1 p-2 px-3 rounded fw-bold" style={{ backgroundColor: frequency.color, color: frequency.textColor }}>{frequency.name}</div>
      ))}
    </div>
  );
}

RouteLegend.propTypes = {
  selectedRouteTypes: PropTypes.array,
};

RouteLegend.defaultProps = {
  selectedRouteTypes: [],
};

export default RouteLegend;
