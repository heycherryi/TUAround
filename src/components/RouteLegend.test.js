/* globals test, expect */

import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import RouteLegend from './RouteLegend';

test('renders RouteLegend', () => {
  const { container } = render(
    <Router>
      <RouteLegend />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
