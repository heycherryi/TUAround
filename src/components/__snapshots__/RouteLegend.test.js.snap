// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RouteLegend 1`] = `
<div>
  <div
    class="d-flex mb-3 justify-content-center flex-wrap"
  >
    <div
      class="mb-1 ms-1 me-1 p-2 px-3 rounded fw-bold"
      style="background-color: rgb(243, 39, 53); color: rgb(255, 255, 255);"
    >
      Frequent
    </div>
    <div
      class="mb-1 ms-1 me-1 p-2 px-3 rounded fw-bold"
      style="background-color: rgb(117, 60, 190); color: rgb(255, 255, 255);"
    >
      Local
    </div>
    <div
      class="mb-1 ms-1 me-1 p-2 px-3 rounded fw-bold"
      style="background-color: rgb(93, 95, 98); color: rgb(255, 255, 255);"
    >
      Connector
    </div>
    <div
      class="mb-1 ms-1 me-1 p-2 px-3 rounded fw-bold"
      style="background-color: rgb(0, 170, 231); color: rgb(255, 255, 255);"
    >
      Express
    </div>
    <div
      class="mb-1 ms-1 me-1 p-2 px-3 rounded fw-bold"
      style="background-color: rgb(0, 30, 97); color: rgb(255, 255, 255);"
    >
      Train Shuttles
    </div>
  </div>
</div>
`;
